File: 701 - Insert into a Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function (root, val) {

    var newNode = new TreeNode(val);

    if (root === null) {
        root = newNode;
    }
    let currentNode = root;
    while (true) {
        if (val < currentNode.val) {
            if (!currentNode.left) {
                currentNode.left = newNode;
                return root;
            } else {
                currentNode = currentNode.left;
            }
        } else if (val > currentNode.val) {
            if (!currentNode.right) {
                currentNode.right = newNode;
                return root;
            } else {
                currentNode = currentNode.right;
            }
        }
    }
    return root;

};

File: Java Solutions/701 - Insert into a Binary Search Tree.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        if (root == null) {
            return new TreeNode(val);
        }
        TreeNode currentNode = root;
        while (true) {
            if (currentNode.val > val) {
                if (currentNode.left == null) {
                    currentNode.left = new TreeNode(val);
                    break;
                } else {
                    currentNode = currentNode.left;
                }
            } else if (currentNode.val < val) {
                if (currentNode.right == null) {
                    currentNode.right = new TreeNode(val);
                    break;
                } else {
                    currentNode = currentNode.right;
                }
            }

        }
        return root;
    }
}